### DISCUSSION TITLE
Mainnet Native Randomness

### DISCUSSION THREAD

**Discussion by kaouki99** [aptos-labs/aptos-developer-discussions#226] [Questions]
### Discord user ID

dawkaxx

### Describe your question in detail.

I deployed a contract on mainnet using the new randomness feature but I get this error:

`Move abort in 0x1::option: EOPTION_NOT_SET(0x40001): The Option is in an invalid state for the operation attempted.
The Option is None while it should be Some.`

The exact same smart contract worked fine on testnet. Do you know if the randomness feature encountered any issue on mainnet? I don't see any event `0x1::randomness::RandomnessGeneratedEvent` on mainnet so I assume the release faced some issues.

Thanks for the help

### What error, if any, are you getting?

`Move abort in 0x1::option: EOPTION_NOT_SET(0x40001): The Option is in an invalid state for the operation attempted.
The Option is None while it should be Some.`

### What have you tried or looked at?  Or how can we reproduce the error?

Tried on testnet before deploying on mainnet

### Which operating system are you using?

Linux (Ubuntu, Fedora, Windows WSL, etc.)

### Which SDK or tool are you using? (if any)

TypeScript SDK

**Comment by asfman** [comment]
The randomness feature is unavailable on the mainnet.
